cpp/src/arrow/dataset/test_util.h

```diff
@@ -157,7 +157,7 @@ class DatasetFixtureMixin : public ::testing::Test {
     std::shared_ptr<RecordBatch> lhs;
     ASSERT_OK(expected->ReadNext(&lhs));
     EXPECT_NE(lhs, nullptr);
-    AssertBatchesEqual(*lhs, batch, true);
+    AssertBatchesEqual(*lhs, batch);
```

Comment:
Why the change here? Behavior is probably not changed for `check_metadata=true`, only for `check_metadata=false`.

Reply:
See https://github.com/apache/arrow/pull/14326#discussion_r998421185, this is only reverting what was added in that other PR (to avoid having that PR be blocked on this one). 
In theory, it shouldn't matter, because if it passes without true, it will certainly pass with `check_metadata=True`. It's just not needed in this case (so to simplify the code and avoid that people wonder why it is here)

Reply:
I see, thank you.